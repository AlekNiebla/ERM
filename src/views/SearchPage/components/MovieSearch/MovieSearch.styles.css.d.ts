export const main: string;
export const label: string;
export const searchLabel: string;
export const searchField: string;
export const optionsLabel: string;
export const options: string;
