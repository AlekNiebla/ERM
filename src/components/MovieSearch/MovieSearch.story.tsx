import * as React from "react";
import { storiesOf } from "@storybook/react";
import MovieSearch from "./MovieSearch";

storiesOf("Components / MovieSearch", module).add("default", () => <MovieSearch />);
