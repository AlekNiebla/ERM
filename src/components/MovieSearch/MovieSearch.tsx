import React from "react";

const MovieSearch = () => <span>MovieSearch</span>;

export default MovieSearch;
