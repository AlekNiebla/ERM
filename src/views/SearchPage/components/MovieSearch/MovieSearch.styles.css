.main {
  display: flex;
  flex-direction: column;
  padding: var(--step-2);
}
:local(.label) {
  composes: Typo-Base from global;
  color: var(--color-gray-dark);
}
.searchLabel {
  composes: label;
  display: block;
  margin-bottom: var(--step-1);
}
.searchField {
  composes: Typo-Small from global;
  color: var(--color-gray-dark);
  border: 1px solid var(--color-gray);
  border-radius: var(--corner-radius);
  padding: var(--step-2);
  margin-bottom: var(--step-2);
}
.optionsLabel {
  composes: label;

  display: block;
  margin-bottom: var(--step-1);
}
.options button {
  margin-bottom: var(--step-1);
  margin-right: var(--step-2);
}
