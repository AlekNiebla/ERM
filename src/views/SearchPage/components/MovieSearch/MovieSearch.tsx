import * as React from "react";
// components
import Button from "Components/Button/Button";
// styles
import * as styles from "./MovieSearch.styles.css";
import { MovieListQuery } from "Root/types";

type searchType = "title" | "genres";

type MoviesearchProps = {
  submitAction?: (query: MovieListQuery) => void;
  query?: MovieListQuery;
};

class MovieSearch extends React.Component<
  MoviesearchProps,
  { searchFieldValue: string; searchBy: searchType }
> {
  constructor(props: {}) {
    super(props);
    this.state = { searchFieldValue: "", searchBy: "title" };

    this.handleSearchFieldChange = this.handleSearchFieldChange.bind(this);
    this.handleSearchByChange = this.handleSearchByChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  componentDidMount() {
    if (this.props.query) {
      this.setState({
        searchFieldValue: this.props.query.search,
        searchBy: this.props.query.searchBy
      });
    }
  }
  handleSearchFieldChange(event: React.ChangeEvent<HTMLInputElement>): void {
    this.setState({ searchFieldValue: event.target.value });
  }
  handleSearchByChange(type: searchType) {
    this.setState({ searchBy: type });
  }
  handleSubmit(event: React.SyntheticEvent): void {
    event.preventDefault();
    this.props.submitAction({
      search: this.state.searchFieldValue,
      searchBy: this.state.searchBy
    });
  }
  render() {
    return (
      <form className={styles.main} onSubmit={this.handleSubmit}>
        <div className={styles.searchWrap}>
          <div className={styles.searchField}>
            <span className={styles.searchLabel}>Find a movie</span>
            <input
              className={styles.searchInput}
              type="text"
              placeholder={"Movie title"}
              onChange={this.handleSearchFieldChange}
              value={this.state.searchFieldValue}
            />
          </div>
          <Button
            label="search"
            variant="white"
            type="submit"
            clickAction={this.handleSubmit}
          />
        </div>
        <div className={styles.options}>
          <span className={styles.optionsLabel}>search by</span>
          <Button
            label="title"
            size="small"
            variant={this.state.searchBy === "title" ? "primary" : "white"}
            clickAction={event => {
              event.preventDefault();
              this.handleSearchByChange("title");
            }}
          />
          <Button
            label="genres"
            size="small"
            variant={this.state.searchBy === "genres" ? "primary" : "white"}
            clickAction={event => {
              event.preventDefault();
              this.handleSearchByChange("genres");
            }}
          />
        </div>
      </form>
    );
  }
}

export default MovieSearch;
