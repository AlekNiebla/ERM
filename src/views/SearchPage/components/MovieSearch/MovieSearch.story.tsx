import * as React from "react";
import { storiesOf } from "@storybook/react";
import MovieSearch from "./MovieSearch";

const containerStyle = {
  backgroundColor: "#1e0f75",
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
  padding: "200px"
};
storiesOf("Components / MovieSearch", module).add("default", () => (
  <div style={containerStyle}>
    <MovieSearch />
  </div>
));
