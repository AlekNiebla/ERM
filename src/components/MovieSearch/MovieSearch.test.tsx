import * as React from "react";
import { shallow } from "enzyme";
// components
import MovieSearch from "./MovieSearch";

describe("MovieSearch component...", () => {
  it("...renders without errors", () => {
    const wrapper = shallow(<MovieSearch />);

    expect(wrapper.find(MovieSearch)).toBeTruthy();
  });
});
