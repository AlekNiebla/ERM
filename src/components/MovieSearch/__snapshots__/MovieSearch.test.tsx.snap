// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearch component... ... matches snapshot 1`] = `ShallowWrapper {}`;

exports[`MovieSearch component... ...renders without errors 1`] = `ShallowWrapper {}`;
